users/serializers.py

```diff
@@ -28,7 +28,7 @@
     """,
     flags=re.I | re.VERBOSE | re.MULTILINE,
 )
-USERNAME_RE_PARTIAL = r"[\w .@_+-]+"
+USERNAME_RE_PARTIAL = r"[ A-Za-z0-9.@_+-]+"
```

Comment:
Although functionality wise it works fine and doesn't allow accented characters, I have two minor things that you might consider here:

1. I'm sure there could be a reference to using the space in the username which is unusual but if we do want to use it we should place that somewhere close to the allowed special characters list e.g. `r"[A-Za-z0-9 .@_+-]+"`
2. Since we are matching the username text through a regex, shouldn't we add this regex validation on the frontend validator(https://github.com/mitodl/mitxonline/blob/main/static/js/lib/validation.js#L20) and we can save us an API call that is just going to tell the users about what characters they can use after hitting the submit button.

Reply:
@arslanashraf7 regarding the point.2, 

- It makes sense to me that there should be `frontend validation` earlier before sending an api call. 
- `space` in username does not make sense to me as well. I keep it by considering it may be a requirement earlier but I have removed it and take input from @gsidebo if `space` really makes sense to be part of `username`.

kindly take a look over another commit.